### ISSUE TITLE
Placeholder for amp-list goes invisible, but still occupies space.

### ISSUE THREAD

**Issue by td234** [ampproject/amphtml#12669]
## What's the issue?
I am using amp-list with both a placeholder and fallback. When list is populated, placeholder his hidden by adding "visibility: none". I believe this would be better if it were "display: none" because it takes up space on the page if you have added images, copy, etc to the placeholder. This is the behavior of fallback (display:none).

**Comment by nainar** [comment]
@aghassemi this also makes sense from a rendering perspective. Most rendering engines have a fast path for elements that have `display:none` set. I doubt that the perf benefits would be perceptible, but I would still vote for the change suggested here and would be happy to make it myself.

**Comment by aghassemi** [comment]
@td234 Do you have an example of where this is causing unexpected behaviour? `placeholders` always fill the `amp-list` and `amp-list` always has a fixed size. So regardless of `display:none` or `visibility:hidden` for placeholder, the `amp-list` component is the one taking up space. 

@nainar I am not sure exactly why but we always `visibility:hidden` placeholders regardless of the component (`amp-list` or anything else). The reason might be that placeholders get hidden after the element loads and `visibility:hidden` won't cause a reflow but `display:none` would. It used to be `display:none` but changed to visibility with this commit: https://github.com/ampproject/amphtml/commit/14f3c75df592da402d5ffeeca59d78c3af6f54e0

**Comment by td234** [comment]
@aghassemi The amp-list takes up space, but at the bottom of the amp-list the placeholder adds additional space between the last list item and the end of the amp-list container. When empty or not shown, the placeholder should not take up any space. It essentially adds extra blank space at the bottom of the amp-list, which it should not for design purposes. For designs where an extra 20 pixels are unwanted, this is a factor. I have modified my CSS to resolve this, so I do not have a working example, but unused elements on a page should be not take up space, in my opinion.

**Comment by aghassemi** [comment]
@td234 Thanks for clarifying. The placeholder should never be able to change the size of its container regardless of `visibility:hidden` or `display:none`. I think that's bug we ought to fix (rather than changing placeholder to become `display:none`.)

I believe (not tested) the root cause pf this behaviour is:
"If a placeholder is `taller` than the final `resized` height of `amp-list`, we `measure` the final height **wrong** by including the placeholder in the calculation. "

**Comment by td234** [comment]
@aghassemi I am pretty sure it is not the behavior as you mention, because if I changed the placeholder in the developer console to display:none, the space was removed. Pretty sure the placeholder itself has some height assigned to it. So making is invisible it still has the height. But, I am sure you know this code far better than I do. Thanks for even considering the change!

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @ericlindley-g Do you have any updates?

**Comment by ericlindley-g** [comment]
/to @aghassemi for now (part of amp-list resizing issues)

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @aghassemi Do you have any updates?

**Comment by cathyxz** [comment]
I think this is solved by https://github.com/ampproject/amphtml/pull/18820: 

```
/* Placeholder, by design, expands layout=container unlike other layouts where
   it is position:absolute and overlays the layout box.
   However, when it goes away it should no longer expand the parent, so instead
   of visibility:hidden it must be display:none */
.i-amphtml-layout-container > [placeholder].hidden,
.i-amphtml-layout-container > [placeholder].amp-hidden {
  display: none;
}
```

Aside from layout container, there shouldn't be any layout where the placeholder ends up adding space to the `amp-list`, since it should just be sized to fit 100% of the amp-list, which will control its own size.

**Comment by aghassemi** [comment]
this should be resolved by #18820, please feel free to reopen if still an issue.
